src/tests/ftest/checksum/csum_error_logging.py

```diff
@@ -19,88 +18,56 @@ class CsumErrorLog(DaosCoreBase):
-    def setUp(self):
-        super().setUp()
-        self.dmg = self.get_dmg_command()
-        self.dmg.hostlist = self.hostlist_servers[0]
 
-    def get_nvme_device_id(self):
-        """method to get nvme device-id. """
-        self.dmg.json.value = True
-        try:
-            result = self.dmg.storage_query_list_devices()
-        except process.CmdError as details:
-            self.fail("dmg command failed: {}".format(details))
-        finally:
-            self.dmg.json.value = False
-
-        data = json.loads(result.stdout_text)
-        resp = data['response']
-        if data['error'] or len(resp['host_errors']) > 0:
-            if data['error']:
-                self.fail("dmg command failed: {}".format(data['error']))
-            else:
-                self.fail("dmg command failed: {}".format(resp['host_errors']))
-        uuid = None
-        for value in list(resp['host_storage_map'].values()):
-            if value['storage']['smd_info']['devices']:
-                uuid = value['storage']['smd_info']['devices'][0]['uuid']
-                break
-        return uuid
-
-    def get_checksum_error_value(self, device_id=None):
+    def get_checksum_error_value(self, dmg, device_id):
         """Get checksum error value from dmg storage_query_device_health.
 
         Args:
+            dmg (DmgCommand): the DmgCommand object used to call storage_query_device_health()
             device_id (str): Device UUID.
-        """
-        if device_id is None:
-            self.fail("No device id provided")
 
-        self.dmg.json.value = True
-        try:
-            result = self.dmg.storage_query_device_health(device_id)
-        except process.CmdError as details:
-            self.fail("dmg command failed: {}".format(details))
-        finally:
-            self.dmg.json.value = False
-
-        data = json.loads(result.stdout_text)
-        resp = data['response']
-        if data['error'] or len(resp['host_errors']) > 0:
-            if data['error']:
-                self.fail("dmg command failed: {}".format(data['error']))
-            else:
-                self.fail("dmg command failed: {}".format(resp['host_errors']))
-        checksum_errs = None
-        for value in list(resp['host_storage_map'].values()):
-            if value['storage']['smd_info']['devices']:
-                dev = value['storage']['smd_info']['devices'][0]
-                checksum_errs = dev['health']['checksum_errs']
-                break
+        Returns:
+            int: the number of checksum errors on the device
+        """
+        checksum_errs = 0
+        info = get_dmg_smd_info(self, dmg.storage_query_device_health, 'devices', uuid=device_id)
+        for devices in info.values():
+            for device in devices:
+                try:
+                    if device['uuid'] == device_id:
+                        checksum_errs = device['health']['checksum_errs']
+                        break
```

Comment:
This `break` only breaks out of the inner `for` loop. Should it break out of the outer as well? If so, maybe a simple
```suggestion
                        return device['health']['checksum_errs']
```

Reply:
Done.